flink-runtime/src/test/java/org/apache/flink/runtime/metrics/util/MetricUtilsTest.java

```diff
@@ -243,4 +240,59 @@ public MetricGroup addGroup(String name) {
                                 METRIC_GROUP_MEMORY,
                                 METRIC_GROUP_MANAGED_MEMORY)));
     }
+
+    // --------------- utility methods and classes ---------------
+
+    /**
+     * An extreme simple class with no dependencies outside "java" package to ease re-defining from
+     * its bytecode.
+     */
+    private static class Dummy {}
+
+    /**
+     * Define an new class using {@link Dummy} class's name and bytecode to consume Metaspace and
+     * NonHeap memory.
+     */
+    @SuppressWarnings("unchecked")
+    private static Class<Dummy> redefineDummyClass() throws ClassNotFoundException {
+        Class<?> clazz = Dummy.class;
+        ChildFirstClassLoader classLoader =
+                new ChildFirstClassLoader(
+                        ClassLoaderUtils.getClasspathURLs(),
+                        clazz.getClassLoader(),
+                        new String[] {"java."},
+                        ignored -> {});
+
+        Class<?> newClass = Class.forName(clazz.getName(), true, classLoader);
```

Comment:
Does this not work?

```suggestion
        Class<?> newClass = classLoader.loadClass(clazz.getName());
```

Reply:
I think both should work for `Dummy`. But `Class.forName(initialize = true)` does not work for `MetricUtils`. I keep it mainly as stringent legacy from `MetricUtils` attempt.

Reply:
It actually does not contribute to "consuming metaspace/non-heap memory". I will do some tests for solely `loadClass`.

Reply:
Hi @zentol, I have changed to `loadClass`, now the test concentrates more on define/load a class but not initialize.